.github/workflows/nightly-test-release.yaml

```diff
@@ -0,0 +1,126 @@
+name: Test release process nightly
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Run every day at midnight (UTC)
+  workflow_dispatch: # Allow running manually on demand
+
+env:
+  RELEASE_TAG: t9.9.9-fake
+
+jobs:
+  nightly-test-release:
+    name: Tag current version for testing
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+      - name: Set and push fake tag for release
+        run: |
+          git tag ${{ env.RELEASE_TAG }}
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          tags: true
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  build-push-services:
+    name: Build, sign and provenance
+    needs: [nightly-test-release]
+    permissions:
+      actions: read
+      contents: write
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        destination: [ghcr]
+        arch: [amd64, arm64, s390x]
+        include:
+        - destination: ghcr
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          image: ghcr.io/rancher-sandbox/rancher-turtles
+          password: GITHUB_TOKEN
+          secret_registry: false
+    uses: ./.github/workflows/release-workflow.yml
+    with:
+      password: ${{ matrix.password }}
+      username: ${{ matrix.username }}
+      registry: ${{ matrix.registry }}
+      tag: t9.9.9-fake
+      arch: ${{ matrix.arch }}
+      image: ${{ matrix.image }}
+      secret_registry: ${{ matrix.secret_registry }}
+    secrets: inherit
+
+  release:
```

Comment:
Should this also have a step which publishes helm chart? And cleanup removes it later.